### ISSUE TITLE
Incremental compilation may be broken for this module

### ISSUE THREAD

**Issue by omus** [invenia/Mocking.jl#5]
As brought up in https://github.com/JuliaTime/TimeZones.jl/issues/30, Mocking.jl reports a warning when pre-compiling a package which uses Mocking:

```
INFO: Recompiling stale cache file /home/tom/.julia/lib/v0.5/CTechCommon.ji for module CTechCommon.
WARNING: eval from module Mocking to CTechCommon:    
Expr(:module, true, :##262, Expr(:block, Expr(:line, 108, :/home/tom/.julia/v0.5/Mocking/src/Mocking.jl)::Any, Expr(:line, 108, :/home/tom/.julia/v0.5/Mocking/src/Mocking.jl)::Any)::Any)::Any
  ** incremental compilation may be broken for this module **

WARNING: replacing module ##262
```

**Comment by omus** [comment]
The solution is to avoid [creating a `PatchEnv`](https://github.com/invenia/Mocking.jl/blob/v0.2.0/src/Mocking.jl#L14) before it is needed. Will be fixed in v0.2.1

**Comment by omus** [comment]
Fixed in https://github.com/invenia/Mocking.jl/commit/fe424677921dadb0384d3480e601a79608810b81
